common/app/com/commercetools/sunrise/shoppingcart/LineItemBeanFactory.java

```diff
@@ -1,22 +1,28 @@
 package com.commercetools.sunrise.shoppingcart;
 
-import com.commercetools.sunrise.common.contexts.UserContext;
 import com.commercetools.sunrise.common.ctp.ProductDataConfig;
-import com.commercetools.sunrise.common.reverserouter.ProductReverseRouter;
+import com.commercetools.sunrise.common.models.ProductAttributeBean;
+import com.commercetools.sunrise.common.models.ProductAttributeBeanFactory;
 import io.sphere.sdk.carts.LineItem;
-import io.sphere.sdk.models.Base;
 
 import javax.inject.Inject;
+import java.util.List;
 
-public class LineItemBeanFactory extends Base {
+import static java.util.stream.Collectors.toList;
+
+public class LineItemBeanFactory extends MiniCartLineItemBeanFactory {
     @Inject
     private ProductDataConfig productDataConfig;
     @Inject
-    private UserContext userContext;
-    @Inject
-    private ProductReverseRouter reverseRouter;
+    private ProductAttributeBeanFactory productAttributeBeanFactory;
 
     public LineItemBean create(final LineItem lineItem) {
-        return new LineItemBean(lineItem, productDataConfig, userContext, reverseRouter);
+        final LineItemBean bean = fillBean(new LineItemBean(), lineItem);
+        final List<ProductAttributeBean> attributes = lineItem.getVariant().getAttributes().stream()
+                .filter(attr -> productDataConfig.getSelectableAttributes().contains(attr.getName()))
+                .map(attr -> productAttributeBeanFactory.create(attr))
+                .collect(toList());
+        bean.setAttributes(attributes);
+        return bean;
```

Comment:
Maybe this can override `fillBean` and call `super()` and then add the attributes. This way it follows the same structure as the other factories.